test/expr/primary/keypath/keypath-objc.swift

```diff
@@ -57,7 +57,7 @@ func testKeyPath(a: A, b: B) {
   let _: String = #keyPath(A.propString)
 
   // Property of String property (which looks on NSString)
-  let _: String = #keyPath(A.propString.URLsInText)
+  let _: String = #keyPath(A.propString.urlsInText)
```

Comment:
It's not clear to me, was this change necessary because of your other changes ? I don't see how.

Reply:
The test started to fail stating the name of the property changed, and I assumed it wasn't the case before because the expression wasn't being indexed, although it does seem weird. Was it not supposed to happen?

Reply:
'Indexing' them should not have any effect in compiler errors, is the case that these were not even typechecked before ?

Reply:
Are you 100% sure your changes are triggering the failure ?

Reply:
You can actually reproduce this in Xcode 10 with old Swift 2 properties (sorry for the iOS example, it's the only one I remember that was a property and not a method):

```
let a = #keyPath(UIColor.redColor)
print(a)
```

It will work and return `redColor` despite the property being called `red` now. You can get it to throw the error if you get the property directly, which is the case where indexing was fine:

```
extension UIColor {
    func getPath() {
        let a = #keyPath(redColor)
        print(a)
    }
}
```

I have not fully located where the `renamed` error comes from but I'm assuming it comes from this change, isn't this where this error would be thrown? https://github.com/apple/swift/blob/df64e8a5dcd96e2b402be4445e0d8e913fedf71b/lib/Sema/TypeCheckAvailability.cpp#L2437

If that's the right place, then this specific problem isn't indexing itself but just that the key paths that included types never stopped being UnresolvedProperties, so this case was never being reached by them.

Reply:
@DougGregor @rudkx please review, with these changes properties from types references in #keypaths are properly typechecked now.